src/group_theory/perm/sign.lean

```diff
@@ -761,3 +777,47 @@ end congr
 end sign
 
 end equiv.perm
+
+section alternating_subgroup
+open equiv.perm
+variables (α) [fintype α] [decidable_eq α]
+
+/-- The alternating group on a finite type, realized as a subgroup of `equiv.perm`.
+  For $A_n$, use `alternating_subgroup (fin n)`. -/
+@[derive fintype] def alternating_subgroup : subgroup (perm α) :=
+sign.ker
+
+instance [subsingleton α] : unique (alternating_subgroup α) :=
+⟨⟨1⟩, λ ⟨p, hp⟩, subtype.eq (subsingleton.elim p _)⟩
+
+variables {α}
+
+lemma alternating_subgroup_eq_sign_ker : alternating_subgroup α = sign.ker := rfl
+
+@[simp]
+lemma mem_alternating_subgroup {f : perm α} :
+  f ∈ alternating_subgroup α ↔ sign f = 1 :=
+sign.mem_ker
+
+lemma prod_list_swap_mem_alternating_subgroup_iff_even_length {l : list (perm α)}
+  (hl : ∀ g ∈ l, is_swap g) :
+  l.prod ∈ alternating_subgroup α ↔ even l.length :=
+begin
+  rw [mem_alternating_subgroup, sign_prod_list_swap hl, ← units.coe_eq_one, units.coe_pow,
+    units.coe_neg_one, nat.neg_one_pow_eq_one_iff_even],
+  dec_trivial
+end
+
+lemma two_mul_card_alternating_subgroup [nontrivial α] :
+  2 * card (alternating_subgroup α) = card (perm α) :=
+begin
+  classical,
+  rw ← fintype.card_units_int,
+  convert (card_eq_card_quotient_mul_card_subgroup (alternating_subgroup α)).symm,
+  convert of_equiv_card (quotient_group.quotient_ker_equiv_of_surjective _
```

Comment:
You want `card_congr` not `of_equiv_card` here - the latter has a bad fintype instance, which is why you need `convert`.

Reply:
I still seem to need `convert` the first time because of different decidability instances.

Reply:
Here's a proof that avoids `convert` and `classical`:
```
lemma two_mul_card_alternating_subgroup [nontrivial α] :
  2 * card (alternating_subgroup α) = card (perm α) :=
begin
  rw [←fintype.card_units_int,
      ←fintype.card_congr
        (quotient_group.quotient_ker_equiv_of_surjective _ (sign_surjective α)).to_equiv],
  exact (card_eq_card_quotient_mul_card_subgroup _).symm,
end
```